python/ray/serve/api.py

```diff
@@ -344,26 +346,35 @@ def deployment(
         Deployment
     """
 
+    # Create list of all user-configured options from keyword args
+    user_configured_option_names = [
+        option
+        for option, value in locals().items()
```

Comment:
maybe add a comment, we should not have any local variable because of using `locals()`? Or Deployment can return you a user-configured attributes.

Reply:
Ah good idea. We can still have local variables, but they should be defined after this list. Creating that list should be the first thing that happens in the function.

Reply:
Yeah, the order is difficult to guarantee and audit, better to have a way to predefine user-configured attributes.

Reply:
I think unit tests should be able to catch any issues with the ordering since any local variables that are defined before this list will show up as a user-configured attribute and cause the unit tests to fail. What do you mean by predefine user-configured attributes?

Reply:
Deployment can provide the user_configured_option_names attributes, you can filter the attributes from `locals()`.